### PR TITLE
Revendor gardener/gardener@v1.12.3, update to golang@1.15.3 and alpine@3.12.1

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area control-plane
/kind cleanup
/priority normal
/platform aws

What this PR does / why we need it:
Vendor gardener/gardener@v1.12.3 and update golang verison to 1.15 and alpine image version to 3.12.1.

```improvement operator
Golang version is updated to `1.15` and alpine image version is updated to `3.12.1`.
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-FROM golang:1.14.9 AS builder
+FROM golang:1.15.3 AS builder
 
 WORKDIR /go/src/github.com/gardener/gardener-extension-provider-aws
 COPY . .
 RUN make install
 
 ############# base
-FROM alpine:3.12.0 AS base
+FROM alpine:3.12.1 AS base
```
Comment (timebertt) [inline Dockerfile:9]: Can you also adapt
https://github.com/gardener/gardener-extension-provider-aws/blob/111ec9999369ae29f2354105b374561895e6c976/charts/gardener-extension-provider-aws/templates/_images.tpl#L2
(image of `mtu-customizer`)?

**Review by timebertt — COMMENTED** [review]
One comment :)

**Review by timebertt — APPROVED** [review]
/lgtm

**Review by DockToFuture — APPROVED** [review]
/lgtm

**Review by dkistner — APPROVED** [review]

**Pull request merged**